includes/payment-methods/class-wc-stripe-express-checkout-helper.php

```diff
@@ -1549,22 +1570,46 @@ public function get_login_confirmation_settings() {
 	/**
 	 * Pages where the express checkout buttons should be displayed.
 	 *
+	 * @param string|null $express_checkout_type The type of express checkout.
 	 * @return array
 	 */
-	public function get_button_locations() {
-		// If the locations have not been set return the default setting.
-		if ( ! isset( $this->stripe_settings['express_checkout_button_locations'] ) ) {
-			return [ 'product', 'cart' ];
+	public function get_button_locations( ?string $express_checkout_type = null ) {
+		switch ( $express_checkout_type ) {
+			case 'amazon_pay':
+				$key = 'amazon_pay_button_locations';
+				break;
+			case 'payment_request':
+			case 'link': // Link does not yet have its own Customize page. It shares the same location settings as Apple Pay and Google Pay.
+			default:
+				$key = 'express_checkout_button_locations';
+				break;
 		}
 
-		// If all locations are removed through the settings UI the location config will be set to
-		// an empty string "". If that's the case (and if the settings are not an array for any
-		// other reason) we should return an empty array.
-		if ( ! is_array( $this->stripe_settings['express_checkout_button_locations'] ) ) {
-			return [];
+		if ( ! isset( $this->stripe_settings[ $key ] ) ) {
+			// If the locations have not been set/modified, return the default setting.
+			$enabled_locations = [ 'product', 'cart' ];
+		} elseif ( ! is_array( $this->stripe_settings[ $key ] ) ) {
+			// If all locations are removed through the settings UI the location config will be set to
+			// an empty string "". If that's the case (and if the settings are not an array for any
+			// other reason) we should return an empty array.
+			$enabled_locations = [];
+		} else {
+			$enabled_locations = $this->stripe_settings[ $key ];
 		}
 
-		return $this->stripe_settings['express_checkout_button_locations'];
+		return $enabled_locations;
+	}
+
+	/**
+	 * Check if the express checkout type is enabled for the given location.
+	 *
+	 * @param string $express_checkout_type The type of express checkout.
+	 * @param string $location The location to check.
+	 * @return boolean
+	 */
+	public function is_enabled_for_location( $express_checkout_type = 'payment_request', $location = '' ) {
```

Comment:
I think we should use an explicit type for `$express_checkout_type`, as `get_button_locations()` uses an explicit type.
```suggestion
	public function is_enabled_for_location( string $express_checkout_type = 'payment_request', string $location = '' ) {
```